src/test/java/cat/udl/eps/softarch/tfgfinder/steps/InviteUserStepDefs.java

```diff
@@ -0,0 +1,184 @@
+                                .accept(MediaType.APPLICATION_JSON).with(AuthenticationStepDefs.authenticate()))
+                .andDo(print());
+    }
+
+    @Then("The invite to user {string} of proposal {string} is created with status {string}")
+    public void theInviteToUserOfProposalShouldBeCreatedWithStatus(String whoUsername, String whatTitle, String status) throws Exception {
+        User who = userRepository.findUserById(whoUsername);
+        Proposal what = proposalRepository.findProposalByTitle(whatTitle);
+        Invite invite = inviteRepository.findByWhoAndWhat(who, what);
+
+        assertEquals(status, invite.getStatus());
+
+        stepDefs.mockMvc.perform(
+                get("/invites/{id}", invite.getId())
+                        .param("whoUsername", whoUsername)
+                        .param("proposalTitle", whatTitle)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(AuthenticationStepDefs.authenticate())
+        ).andExpect(jsonPath("$.status").value(invite.getStatus()));
+    }
+
+    @Then("{string} creates an invite to user {string} for proposal {string} with status {string} and date {string}")
+    public void createsAnInviteToUserForProposalWithStatusAndDate(String ownUsername, String whoUsername, String proposalTitle, String status, String dateString) throws Exception {
+        Proposal proposal = proposalRepository.findProposalByTitle(proposalTitle);
+        ZonedDateTime date = ZonedDateTime.parse(dateString);
+        User who = userRepository.findUserById(whoUsername);
+
+        if ((isStudent(ownUsername) && isStudent(whoUsername)) || (isProfessor(ownUsername) && isProfessor(whoUsername)) || (isExternal(ownUsername) && isExternal(whoUsername))) {
+            return;
+        }
+
+        Invite invite = new Invite();
+        invite.setWho(who);
+        invite.setWhat(proposal);
+        invite.setStatus(status);
+        invite.setInviteDate(date);
+        inviteRepository.save(invite);
+
+        stepDefs.result = stepDefs.mockMvc.perform(
+                        post("/invites")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(stepDefs.mapper.writeValueAsString(invite))
+                                .accept(MediaType.APPLICATION_JSON).with(AuthenticationStepDefs.authenticate()))
+                .andDo(print());
+    }
+
+    private boolean isStudent(String username) {
+        return studentRepository.existsById(username);
+    }
+
+    private boolean isProfessor(String username) {
+        return professorRepository.existsById(username);
+    }
+
+    private boolean isExternal(String username) {
+        return externalRepository.existsById(username);
+    }
+
+    @Then("The invite will not be created")
+    public void the_proposal_should_not_be_created() {
+        Invite invite = new Invite();
```

Comment:
This code is not checking if the invite has not been created... One option would be through the repository that there are no invites, or that when we try to find the invite supposed to be created, the output is an empty list